libmamba/src/core/package_fetcher.cpp

```diff
@@ -325,12 +325,26 @@ namespace mamba
 
     std::string PackageFetcher::channel() const
     {
-        return m_package_info.channel;
+        if (!util::starts_with(m_package_info.package_url, "file://"))
+        {
+            return m_package_info.channel;
+        }
+        else  // local package case
+        {
+            return "";
+        }
     }
 
     std::string PackageFetcher::url_path() const
     {
-        return util::concat(m_package_info.platform, '/', m_package_info.filename);
+        if (!util::starts_with(m_package_info.package_url, "file://"))
+        {
+            return util::concat(m_package_info.platform, '/', m_package_info.filename);
+        }
+        else  // local package case
+        {
+            return m_package_info.package_url;
+        }
```

Comment:
This was added back from https://github.com/mamba-org/mamba/pull/3296/files#r1604585887, but with the right condition this time (local package or not).

Note that `m_package_info.package_type` is always set to `specs::PackageType::Unknown` here (which is why this part of code was initially removed in #3296 in the first place).
And that's because `package_type` is only set in [PackageInfo](https://github.com/mamba-org/mamba/blob/main/libmamba/src/specs/package_info.cpp#L61), and then its value is not passed when going through [make_package_info](https://github.com/mamba-org/mamba/blob/main/libmamba/src/solver/libsolv/helpers.cpp#L99-L113) from `libsolv` so it's always set to Unknown (which is the default).